src/Utils.hh

```diff
@@ -181,6 +182,14 @@ namespace sdf
 
     return errors;
   }
+
+  template <typename T>
+  T *optionalToPointer(std::optional<T> &_shape)
```

Comment:
Should there also be a const overload? `const T *optionalToPointer(const std::optional<T> &_shape)`

Reply:
I actually started with the const version, but there's one class:https://github.com/osrf/sdformat/blob/3ec5f5ae1e2cc3ad59c6f17f5003a6d6737bde1d/src/World.cc#L402 that needs a non-const pointer (not sure if that was intentional). I'll add the overload.

Reply:
Agree, also not to get super picky, but it's not necessarily a "shape" in this call, so maybe the arg could be renamed.

Reply:
Added the overload and changed `shape` to `opt` in https://github.com/osrf/sdformat/pull/472/commits/23721255163308221625f95ec830c6508adc684a